### PR TITLE
docs: Change Assign usage for Name

### PR DESCRIPTION
**Pull request by eriklimakc**
# Must be merged after https://github.com/unoplatform/uno.extensions/pull/2185

## PR Type

What kind of change does this PR introduce?
- Documentation content changes

## What is the new behavior?
Changing `.Assign(out var fe)` usage for `.Name(out var fe)` in docs.

### CODE REVIEW

**Comment by eriklimakc** [comment]
In this second example below, should we leave it as is? If we change it to use `.Name(out var slider)` it would then be exactly the same as the first one.

https://github.com/unoplatform/uno.extensions/blob/071fb9d2b1a8887a13f80cb386002e0c9483e4bc/doc/Learn/Markup/DependencyPropertyBuilder.md?plain=1#L50-L66

cc @Youssef1313 @dansiegel

**Comment by eriklimakc** [comment]
Should this be changed to `.Name(out var myAnimatedRectangle)`?

https://github.com/unoplatform/uno.extensions/blob/071fb9d2b1a8887a13f80cb386002e0c9483e4bc/doc/Learn/Markup/Storyboards.md?plain=1#L23-L25

cc @Youssef1313 @dansiegel

File: doc/Learn/Markup/Binding101.md [hunk 1]
```diff
@@ -147,12 +147,11 @@ public partial class MainPage : Page
...
 			.Content(
 				new TextBox()
-					.Assign(out var searchBox)
-					.Text(() => vm.Query)
-					.Name("SearchBox"),
+					.Name(out var searchBox)
+					.Text(() => vm.Query),
 				new TextBox()
 					.Text(x => x
-						.Source("SearchBox")
```
Comment (Youssef1313) [inline doc/Learn/Markup/Binding101.md:155]: I'm not really sure about this change. Maybe this should be `nameof(...)`

Comment (dansiegel) [reply doc/Learn/Markup/Binding101.md:155]: This code is technically accurate... but we as @Youssef1313 points out not doing `nameof` fundamentally changes what we were demonstrating in the sample code.

Comment (eriklimakc) [reply doc/Learn/Markup/Binding101.md:155]: Thanks guys. So `.Source(nameof(searchBox))` ?

File: doc/Learn/Markup/DependencyPropertyBuilder.md [hunk 2]
```diff
@@ -48,13 +48,17 @@ If you wish to perform tasks such as manipulate the value of the binding or form
 Sometimes you aren't binding to the `DataContext` of element and instead you need to reference another source. With WinUI we have 2 ways of doing this. The first is that we could specify a source directly such as:
```
Comment (agneszitte) [inline doc/Learn/Markup/DependencyPropertyBuilder.md:48]:
```suggestion
Sometimes you aren't binding to the `DataContext` of the element and instead, you need to reference another source. With WinUI we have 2 ways of doing this. The first is that we could specify a source directly such as:
```

File: doc/Learn/Markup/HowTo-CustomMarkupProject-Toolkit.md [hunk 3]
```diff
@@ -72,10 +72,10 @@ The navigation bar can include items such as a back button, a page title, and ot
     ```csharp
     new Button()
 		.Content("Go to Second Page")
-		.Assign(out var navigationButton),
+		.Name(out var navigationButton),
     ```
 
-    The main ideia here is to create a button that have Assign it self the output variable named in this case as navigationButton.
+    The main ideia here is to create a button that have assign itself, through the `Name` extension method, the output variable named in this case as `navigationButton`.
```
Comment (agneszitte) [inline doc/Learn/Markup/HowTo-CustomMarkupProject-Toolkit.md:78]:
```suggestion
    The main idea here is to create a button that has to assign itself, through the `Name` extension method, the output variable named in this case is `navigationButton`.
```

File: doc/Learn/Markup/HowTo-CustomMarkupProject-Toolkit.md [hunk 4]
```diff
@@ -72,10 +72,10 @@ The navigation bar can include items such as a back button, a page title, and ot
...
     After that we need to add a Click EventHandler to add the Navigation using the Frame.Navigate.
```
Comment (agneszitte) [inline doc/Learn/Markup/HowTo-CustomMarkupProject-Toolkit.md:79]:
```suggestion
    After that, we need to add a Click EventHandler to add the Navigation using the Frame.Navigate. 
```

File: doc/Learn/Markup/HowTo-CustomMarkupProject-Toolkit.md [hunk 5]
```diff
@@ -274,18 +274,18 @@ Chips are often used to display short pieces of information such as tags, catego
...
 			
 			new Chip()
 				.Margin(5)
 				.Content("Chip 3")
-				.Assign(out var chipElement)
+				.Name(out var chipElement)
 		)
     ```
 
     And we need to add the event handlers  on our code.
```
Comment (agneszitte) [inline doc/Learn/Markup/HowTo-CustomMarkupProject-Toolkit.md:287]:
```suggestion
    And we need to add the event handlers to our code.
```

File: doc/Learn/Markup/HowTo-CustomMarkupProject-VisualStates.md [hunk 6]
```diff
@@ -345,7 +345,7 @@ With this we can control events and changes.
 		    )
     ```
 
-    We will change the navigationButton (the same that has been Assign for control the Navigation) and will the Background Color now.
+    We will change the navigationButton (the same that has been assigned for controlling the navigation) and change its background color now.
```
Comment (agneszitte) [inline doc/Learn/Markup/HowTo-CustomMarkupProject-VisualStates.md:348]:
```suggestion
    We will change the navigationButton (the same one that has been assigned for controlling the navigation) and change its background color now.
```

**Comment by kazo0** [comment]
@Mergifyio backport release/stable/4.1

**Review by Youssef1313 — APPROVED** [review]

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**